spring-context/src/main/java/org/springframework/context/annotation/PropertySourceRegistry.java

```diff
@@ -64,7 +64,7 @@ void processPropertySource(AnnotationAttributes propertySource) throws IOExcepti
 			encoding = null;
 		}
 		String[] locations = propertySource.getStringArray("value");
-		Assert.isTrue(locations.length > 0, "At least one @PropertySource(value) location is required");
+		Assert.notEmpty(locations, "At least one @PropertySource(value) location is required");
```

Comment:
This is unnecessary since `org.springframework.core.annotation.AnnotationAttributes.getStringArray(String)` will never return `null`.